### ISSUE TITLE
Kore label munging confusion

### ISSUE THREAD

**Issue by ehildenb** [runtimeverification/k#3023]
According to the munging table in [StringUtil](https://github.com/runtimeverification/k/blob/master/kore/src/main/java/org/kframework/utils/StringUtil.java#L551), and also to [KoreToK](https://github.com/runtimeverification/k/blob/fc61802320bf64d4e0b1f2462e7e91ea2d3296bd/kore/src/main/scala/org/kframework/parser/kore/parser/KoreToK.scala#L37), the munging of `-` should be `'Hyph'`. But I take this module in file `test.k`:

```
module MY-MODULE
  imports BOOL
  configuration <k> $PGM:Pgm </k>

  syntax Pgm ::= "foo"
endmodule
```

And it produces this Kore with `kompile --backend haskell test.k --main-module MY-MODULE`, and I inspect the `definition.kore`, and see this symbol present:

```
symbol Lblfoo'Unds'MY-MODULE'Unds'Pgm{} ...
```

I guess instead it should be:

```
symbol Lblfoo'Unds'MY'Hyph'MODULE'Unds'Pgm{} ...
```

I discovered this trying to reproduce `kast_to_kore` and `kore_to_kast` here: https://github.com/runtimeverification/pyk/pull/73

**Comment by ehildenb** [comment]
It appears [that here in ModuleToKORE](https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java#L1851) we are overriding the munge table for a few specific symbols, in particular we're undoing the `-` munging.

Maybe the `StringUtil` and `KoreToK` implementations should be updated to just not munge `-` in the first place? We don't use the OCAML backend anymore, and I guess that was the reason for this particular case.

**Comment by radumereuta** [comment]
Remove the hyphen from the munging table.

**Comment by Baltoli** [comment]
We should check whether this has been fixed. If not, we should go ahead and remove it from the table.

**Comment by Robertorosmaninho** [comment]
This is still happening, I'll address this issue in a new PR.

**Comment by Robertorosmaninho** [comment]
Changes on the llvm backend will be necessary.

**Comment by Robertorosmaninho** [comment]
Closed with #3401
